osquery/core/query.cpp

```diff
@@ -42,12 +42,19 @@ uint64_t Query::getPreviousEpoch() const {
   return epoch;
 }
 
-uint64_t Query::getQueryCounter(bool new_query) const {
+uint64_t Query::getQueryCounter(bool all_records, bool new_query) const {
   uint64_t counter = 0;
-  if (new_query) {
+  if (all_records) {
     return counter;
   }
```

Comment:
I can roll with this, but I think it's a bit confusing.

For events, this calls: `incrementCounter(false, new_epoch || new_query, counter);`
but for normal queries:  `incrementCounter(new_epoch, new_query, counter);`

And that distinction feels weird. I don't really understand why it's not something like:
```
uint64_t Query::getQueryCounter(bool start_at_zero, bool new_query) const {
  if (new_query) {
    if (all_records) {
      return 0;
    }
    return 1;
}
    uint64_t counter = 0;

```

which would allow a more consistent call semantics. 

But I might be missing something?

Reply:
Yeah I found making a clear distinction between if we are returning all records vs something about the query changed was the least confusing option for me, but I agree it's tricky.

The issue is that events and normal queries are different in that events never have a snapshot/all records available, so they never start at 0 and always reset to 1, while normal queries have a distinction between epoch change or just query SQL changing when they are resetting the counter (to 0 vs 1). 

So normal queries would still need to change start_at_zero based on `new_epoch`. ie: `incrementCounter(new_epoch, new_query || new_epoch, counter)`, which means it doesn't really change the logic in a meaningful way.